### ISSUE TITLE
Can not compile on IBM Blue Gene (ALCF-Mira)

### ISSUE THREAD

**Issue by felixit** [libMesh/libmesh#1391]
I am trying to install the libmesh on the IBM Blue Gene machine on ALCF-Mira. I am using the gcc 4.8.4. The ./configure works fine, but there are several errors during the make process. I attach both of the configure log and the make log files.

**Below is the configure log file:**

[config.log.txt](https://github.com/libMesh/libmesh/files/1213007/config.log.txt)

**Below is the make log file:**
`Making all in include
make[1]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
make  all-recursive
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
Making all in libmesh
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make  all-am
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[4]: Nothing to be done for `all-am'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[1]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
Making all in contrib
make[1]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib'
Making all in boost/include
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/boost/include'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/boost/include'
Making all in eigen/eigen
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/eigen/3.2.9'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/eigen/3.2.9'
Making all in gmv
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/gmv'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/gmv'
Making all in netcdf/v4
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make  all-recursive
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
Making all in include
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/include'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/include'
Making all in libdispatch
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libdispatch'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libdispatch'
Making all in libsrc
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libsrc'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libsrc'
Making all in liblib
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/liblib'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/liblib'
Making all in ncgen3
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/ncgen3'
  CCLD     ncgen3
/soft/compilers/gcc/4.8.4/lib/gcc/powerpc64-bgq-linux/4.8.4/../../../../powerpc64-bgq-linux/bin/ld: attempted static link of dynamic object `../liblib/.libs/libnetcdf.so'
collect2: error: ld returned 1 exit status
make[4]: *** [ncgen3] Error 1
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/ncgen3'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib'
make: *** [all-recursive] Error 1`

**Comment by jwpeterson** [comment]
Looks like there's only one error:
```
attempted static link of dynamic object../liblib/.libs/libnetcdf.so
```

If I recall correctly, on these machines you have to pass the `--enable-all-static` to configure to force libtool to link against static libraries. This was the fix we discovered in cd4ae6d2270 anyway.

**Comment by felixit** [comment]
Hi jwpeterson,

Thank you for your help. However, after I use --enable-all-static to configure, the error still exists. I attach my configure.log and the make.log file:
[config.log.txt](https://github.com/libMesh/libmesh/files/1215660/config.log.txt)

Make.log:
Making all in include
make[1]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
make  all-recursive
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
Making all in libmesh
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make  all-am
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[4]: Nothing to be done for `all-am'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[1]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
Making all in contrib
make[1]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib'
Making all in boost/include
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/boost/include'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/boost/include'
Making all in eigen/eigen
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/eigen/3.2.9'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/eigen/3.2.9'
Making all in gmv
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/gmv'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/gmv'
Making all in netcdf/v4
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make  all-recursive
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
Making all in include
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/include'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/include'
Making all in libdispatch
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libdispatch'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libdispatch'
Making all in libsrc
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libsrc'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libsrc'
Making all in liblib
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/liblib'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/liblib'
Making all in ncgen3
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/ncgen3'
  CCLD     ncgen3
/soft/compilers/gcc/4.8.4/lib/gcc/powerpc64-bgq-linux/4.8.4/../../../../powerpc64-bgq-linux/bin/ld: attempted static link of dynamic object `../liblib/.libs/libnetcdf.so'
collect2: error: ld returned 1 exit status
make[4]: *** [ncgen3] Error 1
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/ncgen3'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib'
make: *** [all-recursive] Error 1

**Comment by pbauman** [comment]
Is it --enable-all-static or --enable-static? Can you run configure --help
and make sure you're using the correct option.

**Comment by jwpeterson** [comment]
Oh, now I remember that you're right, we might need to pass all three of `--disable-shared --enable-static --enable-all-static` on this system. If you don't explicitly opt out of building shared libraries, I think the default is to still build them, and the "all-static" business really just passes a flag to libtool, it doesn't control what libraries are built for libmesh. 

Sorry for the confusion...

**Comment by felixit** [comment]
Hi Bauman and Peterson,

I have tried the --disable-shared --enable-static --enable-all-static, but still it can not compile correctly. The error seems the same as before.
[config.log.txt](https://github.com/libMesh/libmesh/files/1216362/config.log.txt)

Here is the **make.log** file:
`Making all in include
make[1]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
make  all-recursive
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
Making all in libmesh
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make  all-am
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[4]: Nothing to be done for `all-am'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include/libmesh'
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
make[1]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/include'
Making all in contrib
make[1]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib'
Making all in boost/include
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/boost/include'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/boost/include'
Making all in eigen/eigen
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/eigen/3.2.9'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/eigen/3.2.9'
Making all in gmv
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/gmv'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/gmv'
Making all in netcdf/v4
make[2]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make  all-recursive
make[3]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
Making all in include
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/include'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/include'
Making all in libdispatch
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libdispatch'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libdispatch'
Making all in libsrc
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libsrc'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/libsrc'
Making all in liblib
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/liblib'
make[4]: Nothing to be done for `all'.
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/liblib'
Making all in ncgen3
make[4]: Entering directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/ncgen3'
  CCLD     ncgen3
/soft/compilers/gcc/4.8.4/lib/gcc/powerpc64-bgq-linux/4.8.4/../../../../powerpc64-bgq-linux/bin/ld: attempted static link of dynamic object `../liblib/.libs/libnetcdf.so'
collect2: error: ld returned 1 exit status
make[4]: *** [ncgen3] Error 1
make[4]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1/ncgen3'
make[3]: *** [all-recursive] Error 1
make[3]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make[2]: *** [all] Error 2
make[2]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib/netcdf/4.4.1.1'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/gpfs/mira-home/fljia/library/libmesh/contrib'
make: *** [all-recursive] Error 1`

**Comment by jwpeterson** [comment]
Hmm... and you did `rm -rf build` before trying the new configure options, right?

**Comment by felixit** [comment]
Hi Peterson,

Thank you for your reminding. I remove everything and git clone from the source again. Then I use --disable-shared --enable-static --enable-all-static to configure. Then I found the compiling process can go further, but still fails. The make.log file is a little large, so I attach both the configure.log and the make.log. Looks like the Blue Gene machines are hard to handle. 
[config.log.txt](https://github.com/libMesh/libmesh/files/1216693/config.log.txt)
[make.log.txt](https://github.com/libMesh/libmesh/files/1216694/make.log.txt)

**Comment by jwpeterson** [comment]
Looks like the problem is that the fparser JIT compiler is using dlopen, which requires shared library support. 
```
./.libs/libmesh_dbg.a(libdbg_la-fparser_ad.o): In function `FunctionParserADBase<double>::JITCompileHelper(std::string const&)':
/gpfs/mira-home/fljia/library/libmesh/contrib/fparser/fparser_ad.cc:668: warning: Using 'dlopen' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
```
It looks like our test for `dlopen` (in m4/dlopen.m4) is insufficient, we are somehow not respecting the `--disable-shared` flags when doing the compiler test. 

Furthermore, it does not appear to be possible to independently disable the JIT stuff-I do not see an `AC_ARG_ENABLE` call specifically for this, so your best immediate fix is to configure with `--disable-fparser` on your system.

**Comment by jwpeterson** [comment]
I believe the remaining `--enable-all-static` build issues have been fixed by #1403, but if you encounter more on MIRA feel free to reopen.
